### PR TITLE
feat(get): get company certificate types

### PR DESCRIPTION
**Pull request by AnuragNagpure**
## Description

Static Data Endpoint to retrieve supported certificate types

## Why

Suggested API Endpoint: GET /api/administration/staticdata/certificateTypes

## Issue

#469

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/administration/Administration.Service/Controllers/StaticDataController.cs [hunk 1]
```diff
@@ -104,4 +104,19 @@ public StaticDataController(IStaticDataBusinessLogic logic)
...
+    /// </summary>
+    /// <returns>AsyncEnumerable of certificate types</returns>
+    /// <remarks>
+    /// Example: GET: /api/administration/staticdata/certificateTypes    
+    /// </remarks>
+    /// <response code="200">Returns a list of all of the Language i.e german and english</response>
+    [HttpGet]
+    [Authorize(Roles = "view_certificates")]
+    [Route("certificateTypes")]
+    [ProducesResponseType(typeof(IAsyncEnumerable<CompanyCertificateTypeData>), StatusCodes.Status200OK)]
```
Comment (Phil91) [inline src/administration/Administration.Service/Controllers/StaticDataController.cs:119]: Please change to IEnumerable, Swagger UI has problems with displaying IAsyncEnumerable as a ResponseType. 

```suggestion
    [ProducesResponseType(typeof(IEnumerable<CompanyCertificateTypeData>), StatusCodes.Status200OK)]
```

File: src/portalbackend/PortalBackend.DBAccess/Models/CompanyCertificateTypeData.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/********************************************************************************
+ * Copyright (c) 2021, 2023 Contributors to the Eclipse Foundation
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Models/CompanyCertificateTypeData.cs:2]: Please adjust to 

```suggestion
 * Copyright (c) 2024 Contributors to the Eclipse Foundation
```

File: src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs [hunk 3]
```diff
@@ -101,4 +101,17 @@ public StaticDataRepository(PortalDbContext portalDbContext)
         _dbContext.LicenseTypes.AsNoTracking()
             .Select(licenseType => new LicenseTypeData((int)licenseType.Id, licenseType.Label))
             .AsAsyncEnumerable();
+
+    ///<inheritdoc />        
+    public IAsyncEnumerable<CompanyCertificateTypeData> GetCertificateTypes() =>
+        _dbContext.CompanyCertificateTypes.AsNoTracking()
+             .Select(certificate => new CompanyCertificateTypeData
+                (
+                    certificate.Id.ToString(),
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs:110]: The CompanyCertificateId and Label are the same in this case, do we really need this?

**Review by Phil91 — CHANGES_REQUESTED** [review]
Please add a controller test as well

**Comment by Phil91** [comment]
One minor thing I recognised. Please always add @ntruchsess as a Reviewer as well

File: src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs (hunk 3 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs:110]: @jjeroch Do we need company_certificate_type.Id as only label is required to in frontend ?

cc @Phil91

**Comment by AnuragNagpure** [comment]
> Please add a controller test as well

controller test added

**Comment by AnuragNagpure** [comment]
> One minor thing I recognised. Please always add @ntruchsess as a Reviewer as well

Noted

File: src/portalbackend/PortalBackend.DBAccess/Models/CompanyCertificateTypeData.cs (hunk 2 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.DBAccess/Models/CompanyCertificateTypeData.cs]: added

File: src/administration/Administration.Service/Controllers/StaticDataController.cs (hunk 1 above)
Comment (AnuragNagpure) [reply src/administration/Administration.Service/Controllers/StaticDataController.cs]: replaced with IEnumerable

**Comment by Phil91** [comment]
I would wait for a response from Julia, but implementation wise the PR is ready to be approved and merged in my opinion

File: src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs [hunk 4]
```diff
@@ -101,4 +101,17 @@ public StaticDataRepository(PortalDbContext portalDbContext)
         _dbContext.LicenseTypes.AsNoTracking()
             .Select(licenseType => new LicenseTypeData((int)licenseType.Id, licenseType.Label))
             .AsAsyncEnumerable();
+
+    ///<inheritdoc />        
+    public IAsyncEnumerable<CompanyCertificateTypeData> GetCertificateTypes() =>
+        _dbContext.CompanyCertificateTypes.AsNoTracking()
+             .Select(certificate => new CompanyCertificateTypeData
+                (
+                    (int)certificate.Id,
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs:110]: no cast here, please use the enum type as it is. There is no overhead. EF will query the int and pass on the enum-type.

File: src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs [hunk 5]
```diff
@@ -101,4 +101,17 @@ public StaticDataRepository(PortalDbContext portalDbContext)
...
+                    certificate.Label,
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs:111]: Label is redundant as this is defined in code being the string-value of the enum. Furthermore the controller will do the convertion to String on-the-fly, so usually one never needs to do an explicit convertion within the custom-code.

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs (hunk 4 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs]: Casting removed

File: src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs (hunk 5 above)
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs]: Label field is removed as both Id and label are same

File: src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs [hunk 6]
```diff
@@ -101,4 +101,17 @@ public StaticDataRepository(PortalDbContext portalDbContext)
...
+                    certificate.CompanyCertificateTypeDescriptions
+                    .Where(y => y.CompanyCertificateTypeId == certificate.Id)
```
[the remark this answers is not in the document]
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs]: filter with active status

File: src/portalbackend/PortalBackend.DBAccess/Models/CompanyCertificateTypeData.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.DBAccess.Models;
+
+/// <summary>
+/// Model for CompanyCertificateTypeData
+/// </summary>]
+/// <param name="companyCertificateId">CompanyCertificate Type Id</param>
+/// <param name="name">CompanyCertificate Type Name</param>
+/// <param name="description">description</param>
+/// <returns></returns>
+public record CompanyCertificateTypeData(int companyCertificateId, string name, IEnumerable<CertificateTypeDescription> description);
```
[the remark this answers is not in the document]
Comment (AnuragNagpure) [reply src/portalbackend/PortalBackend.DBAccess/Models/CompanyCertificateTypeData.cs]: companyCertificateId changed into enum type

File: src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs (hunk 3 above)
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/StaticDataRepository.cs]: I had a talk with Julia and adjusted the endpoint accodingly. We only include the enum-value being automatically translated to the string (which is the same as the label in the database).

**Pull request merged**